### PR TITLE
ICM: Feature/add helm chart unit tests (#139)

### PR DESCRIPTION
**Pull request by skiesewetter-intershop**
## PR Type

[x] Build-related changes

## Release ##

Be sure to include PR label `major`, `minor` or `patch` when merging into `main`. This determines which part of the semantic version number needs to be bumped automatically.

## What Is the Current Behavior?

Helm chart is checked with lint and integration tested trying an install but not unit tested.

## What Is the New Behavior?

Helm chart is checked with lint and integration tested trying an install and unit tested.

## Does this PR Introduce a Breaking Change?

[ ] Yes
[x] No

### CODE REVIEW

File: charts/icm-as/tests/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# README for helm unit tests
+
+The files contained inside the same folder as this `README.md` contain test suite/case descriptors that can be executed using a helm plugin called [unittest](https://github.com/quintush/helm-unittest).
+
+## Test structure
+
+Each yaml file contains a test suite as defines by the [unittest]https://github.com/quintush/helm-unittest/blob/master/DOCUMENT.md#test-suite plugin.
```
Comment (khauser) [inline charts/icm-as/tests/README.md:7]: "as defined" or better "is based on"

File: charts/icm-as/tests/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+The test suites are structured as follows:
+* `default-values_test.yaml` just checks if the chart can be renders using the checked in `values.yaml` (no custom values set).
```
Comment (khauser) [inline charts/icm-as/tests/README.md:9]: "rendered" and "(with no custom values set)"

File: charts/icm-as/tests/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# README for helm unit tests
+
+The files contained inside the same folder as this `README.md` contain test suite/case descriptors that can be executed using a helm plugin called [unittest](https://github.com/quintush/helm-unittest).
+
+## Test structure
+
+Each yaml file contains a test suite as defines by the [unittest]https://github.com/quintush/helm-unittest/blob/master/DOCUMENT.md#test-suite plugin.
+The test suites are structured as follows:
+* `default-values_test.yaml` just checks if the chart can be renders using the checked in `values.yaml` (no custom values set). 
+* there's a suite for each aspect (section) of the values
```
Comment (khauser) [inline charts/icm-as/tests/README.md:10]: "of the `values.yaml`"

File: charts/icm-as/tests/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+The files contained inside the same folder as this `README.md` contain test suite/case descriptors that can be executed using a helm plugin called [unittest](https://github.com/quintush/helm-unittest).
+
+## Test structure
+
+Each yaml file contains a test suite as defines by the [unittest]https://github.com/quintush/helm-unittest/blob/master/DOCUMENT.md#test-suite plugin.
+The test suites are structured as follows:
+* `default-values_test.yaml` just checks if the chart can be renders using the checked in `values.yaml` (no custom values set). 
+* there's a suite for each aspect (section) of the values 
+    * example: `customizations_test.yaml`
+* there's a suite for all the _basic_  values (or value that are just passed through) named `basic-config-values_test.yaml`
```
Comment (khauser) [inline charts/icm-as/tests/README.md:12]: whitespace after _basic_

File: charts/icm-as/tests/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+## Execution
+
+Just execute 
+
+    helm unittest --helm3 .
```
Comment (khauser) [inline charts/icm-as/tests/README.md:18]: `helm unittest --helm3 .`

File: charts/icm-as/tests/README.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+inside a shell. 
+
+## Plugin installation
+
+Just execute 
+
+    helm plugin install https://github.com/quintush/helm-unittest
```
Comment (khauser) [inline charts/icm-as/tests/README.md:26]: `helm plugin install https://github.com/quintush/helm-unittest`

File: charts/icm-as/values.yaml [hunk 7]
```diff
@@ -5,21 +5,21 @@
 replicaCount: 1
 
 # defined where the application shall run on
```
Comment (khauser) [inline charts/icm-as/values.yaml:7]: define

File: charts/icm-as/tests/README.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
+# README for helm unit tests
+
+The files contained inside the same folder as this `README.md` contain test suite/case descriptors that can be executed using a helm plugin called [unittest](https://github.com/quintush/helm-unittest).
```
Comment (khauser) [inline charts/icm-as/tests/README.md:3]: "The files relative to this ``README.md`" would be much shorter.

**Review by khauser — CHANGES_REQUESTED** [review]
when we squash commit it into `develop/icm` the commit should start with `feat(icm): ...`

**Review by khauser — CHANGES_REQUESTED** [review]
There are some comments...

**Review by khauser — DISMISSED** [review]

**Review by khauser — APPROVED** [review]

**Review by khauser — DISMISSED** [review]

**Review by jometzner — APPROVED** [review]

**Review by stefanh1 — APPROVED** [review]

File: charts/icm-as/templates/as-deployment.yaml [hunk 9]
```diff
@@ -49,10 +49,6 @@ spec:
         - name: icm-as-server
           image: "{{ .Values.image.repository }}"
           imagePullPolicy: "{{ .Values.image.pullPolicy }}"
-          command:
```
Comment (khauser) [inline charts/icm-as/templates/as-deployment.yaml:52]: We still need this for the ISTE test execution. As far as I know there is no entrypoint field in k8s. So this command would overwrite the ENTRYPOINT in the Dockerfile.

**Review by khauser — CHANGES_REQUESTED** [review]

File: charts/icm-as/values.yaml [hunk 10]
```diff
@@ -28,8 +28,6 @@ image:
   pullPolicy: IfNotPresent
   # 'appServer' container provides the application / service endpoint
   repository: intershophub/icm-as:11.0.5
-  command: |
```
Comment (khauser) [inline charts/icm-as/values.yaml:31]: If an `if`-condition would be used for the icm-as.image.command block this would be correct here.

**Review by khauser — APPROVED** [review]

**Pull request merged**